### ISSUE TITLE
socketutils failures on Windows

### ISSUE THREAD

**Issue by kalefranz** [mahmoud/boltons#73]
Another follow-on from #71 

The `tests\test_socketutils.py` file has 9 failures on Windows python 2.7 and 3.4, and 2 failures on Windows python 3.5.

python 2.7: https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.2072/job/8e0vidh6my09nlrt
python 3.4: https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.2072/job/4tw9568trk76v9mp
python 3.5: https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.2072/job/uy77xdh8x8qlwgxg

Again, I'll dig in on a PR at some point if somebody else doesn't beat me to it.

**Comment by kalefranz** [comment]
As a side note, @mahmoud would you be interested in adding Appveyor to your CI to cover interacting with the OS-level stuff on Windows?  I can add the `appveyor.yml` to a PR, but you'll probably want to set up the Appveyor account yourself for the project.

**Comment by mahmoud** [comment]
First of all, thank you for all this great work. Good to have expert eyes on this.

Yeahhh, socketutils is a newly exposed module without good test coverage on windows (though I should have marked them as such). Windows doesn't have socketpair, which is a dang old shame. I was planning on seeing if I could ctypes or otherwise fake a socketpair on windows. 

TLDR the test suite doesn't pass on windows yet, but I did run a big old regression (off-camera), so socketutils itself is OK for Windows, I believe.

And yes, Appveyor is seeming like a better and better idea. I'll see about getting that done by tonight.

**Comment by kalefranz** [comment]
For appveyor, since you use tox already, https://github.com/audreyr/cookiecutter is probably a good resource to emulate.  It was for me at least.

**Comment by mahmoud** [comment]
Cool, yeah, cookiecutter looks good enough for boltons. I was checking out appveyor a bit yesterday and seeing some weird errors, but it seems they were transient, as I'd hoped. I'll keep you posted here!

**Comment by mahmoud** [comment]
Hey @kalefranz this weekend [I added appveyor for boltons](https://ci.appveyor.com/project/mahmoud/boltons). Compared to running it on my old Windows laptop I have lying around, there are still some quirks, for the record:

* It seems os.path.expanduser does nothing
* The getpass module is also hobbled ([which is why the current build is failing](https://ci.appveyor.com/project/mahmoud/boltons/build/1.0.12/job/pvk6yhfhg8fbj3d4#L113))

I doubt these issues would manifest for the vast majority of users, but I'll address them anyways in coming commits. Thanks again for the great info!
